packages/compass-assistant/src/docs-provider-transport.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  type ChatRequestOptions,
+  type ChatTransport,
+  type UIMessage,
+  type UIMessageChunk,
+  convertToModelMessages,
+  streamText,
+} from 'ai';
+import { createOpenAI } from '@ai-sdk/openai';
+
+const openai = createOpenAI({
+  baseURL: 'https://knowledge.staging.corp.mongodb.com/api/v1',
```

Comment:
This might not be using atlas service directly for fetching, but I think we should still at least have this url as part of `AtlasServiceConfig` (and so similarly dynamically accessed from it)

Reply:
You might need to reshuffle components a bit to make sure that atlas service is above this one, but otherwise yeah: use a locator, get a config value from it in activate, pass it to the transport (a bit of an advanced pattern: if you want to create a "service" for the transport, you can package all this logic inside of it, you can look at some usage of `createServiceProvider` method for something like this, but also no need to complicate it too much, just an idea)